ngramoffset.go

```diff
@@ -112,3 +122,211 @@ func (a *arrayNgramOffset) DumpMap() map[ngram]simpleSection {
 func (a *arrayNgramOffset) SizeBytes() int {
 	return 8*len(a.tops) + 4*len(a.bots) + 4*len(a.offsets)
 }
+
+// combinedNgramOffset combines an ascii ngram mapping with a unicode ngram mapping,
+// falling back on unicode for unicode ngrams or ascii ngrams with excessive lengths.
+type combinedNgramOffset struct {
+	asc *asciiNgramOffset
+	uni *arrayNgramOffset
+}
+
+func makeCombinedNgramOffset(ngrams []ngram, offsets []uint32) combinedNgramOffset {
+	// split ngrams & offsets into ascii ngrams and unicode ngrams,
+	// since ascii ngrams can be represented much more compactly (21b instead of 63b)
+	ngramsAscii := make([]uint32, 0, len(ngrams)*11/10)
+	offsetsAscii := make([]uint32, 0, len(ngrams)*11/10)
+
+	ngramsUnicode := make([]ngram, 0, len(ngrams)*2/10)
+	offsetsUnicode := make([]uint32, 0, len(ngrams)*2/10)
```

Comment:
are these caps estimates based on the data in your test corpus? How did you calculate them?